tenant/service_test.go

```diff
@@ -84,8 +87,35 @@ func (s *TenantServiceTestSuite) TestUpdateTenant() {
 		tenant := fxt.Tenants[0]
 		// when
 		tenant.Profile = "unknown"
-		err := svc.CreateOrUpdateTenant(tenant)
+		err := svc.SaveTenant(tenant)
 		// then
 		assert.Error(t, err)
 	})
 }
+
+func (s *TenantServiceTestSuite) TestLookupTenantByNamespace() {
+	s.T().Run("ok", func(t *testing.T) {
+		// given
+		fxt := testfixture.NewTestFixture(s.T(), s.DB, testfixture.Tenants(1), testfixture.Namespaces(1))
```

Comment:
Technically `fxt := testfixture.NewTestFixture(s.T(), s.DB, testfixture.Namespaces(1))` is enough due to the dependency on `Tenants` right?

Btw. I would import the testfixture as `tf` for more readability.

Also, you must not pass `s.T()` but `t` here and in all other places as well. Failing an outer test from the inside is really hard to debug.

Reply:
ah right, I had changed a few `t` / `s.T()` as I remembered you already warned me about this bug, but I guess I forgot a few places :/ And yes, `fxt := testfixture.NewTestFixture(s.T(), s.DB, testfixture.Namespaces(1))` should be enough, but I wanted to be explicit on the fact that we need a `Tenant` because it's what we're actually searching in this test.